plugin/pkg/scheduler/schedulercache/node_info_test.go

```diff
@@ -0,0 +1,370 @@
+					Weight: 5,
+					PodAffinityTerm: v1.PodAffinityTerm{
+						LabelSelector: &metav1.LabelSelector{
+							MatchExpressions: []metav1.LabelSelectorRequirement{
+								{
+									Key:      "security",
+									Operator: metav1.LabelSelectorOpIn,
+									Values:   []string{"S1"},
+								},
+							},
+						},
+						TopologyKey: "region",
+					},
+				},
+			},
+		},
+	}
+	podWithAffinity := makePod("foo", "", &affinity)
+	podWithoutAffinity := makePod("bar", "", &v1.Affinity{})
+	newNodeInfo := NewNodeInfo([]*v1.Pod{podWithAffinity, podWithoutAffinity}...)
+
+	nodeInfo := NodeInfo{
+		allowedPodNumber:    0,
+		requestedResource:   &Resource{},
+		allocatableResource: &Resource{},
+
+		pods:             []*v1.Pod{podWithAffinity, podWithoutAffinity},
+		podsWithAffinity: []*v1.Pod{podWithAffinity},
+		generation:       2,
+		nonzeroRequest:   &Resource{MilliCPU: 2 * priorityutil.DefaultMilliCpuRequest, Memory: 2 * priorityutil.DefaultMemoryRequest},
+	}
+	if !reflect.DeepEqual(*newNodeInfo, nodeInfo) {
+		t.Fatalf("newnodeinfo error,expected: %v,got: %v", nodeInfo, *newNodeInfo)
+	}
+}
+func TestNode(t *testing.T) {
+	node := makeNode("testnode", map[string]string{})
+	nodeInfo := makeNodeInfo(node, []v1.Taint{})
+	if !reflect.DeepEqual(nodeInfo.Node(), node) {
+		t.Fatalf("get nodeinfo node error,expected: %v,got: %v", node, nodeInfo.Node())
+	}
+}
+
+func TestPodsWithAffinity(t *testing.T) {
+	var nodeInfo *NodeInfo
+	affinity := v1.Affinity{
+		PodAffinity: &v1.PodAffinity{
+			PreferredDuringSchedulingIgnoredDuringExecution: []v1.WeightedPodAffinityTerm{
+				{
+					Weight: 5,
+					PodAffinityTerm: v1.PodAffinityTerm{
+						LabelSelector: &metav1.LabelSelector{
+							MatchExpressions: []metav1.LabelSelectorRequirement{
+								{
+									Key:      "security",
+									Operator: metav1.LabelSelectorOpIn,
+									Values:   []string{"S1"},
+								},
+							},
+						},
+						TopologyKey: "region",
+					},
+				},
+			},
+		},
+	}
+	if !reflect.DeepEqual(nodeInfo.PodsWithAffinity(), []*v1.Pod(nil)) {
+		t.Fatalf("get nodeinfo node error,expected: %#v,got: %#v", []*v1.Pod(nil), nodeInfo.PodsWithAffinity())
+	}
+
+	node := makeNode("testnode", map[string]string{})
+	nodeInfo = makeNodeInfo(node, []v1.Taint{})
+	podWithoutAffinity := makePod("foo", "", &v1.Affinity{})
+	podWithAffinity := makePod("foo", "", &affinity)
+	nodeInfo.addPod(podWithAffinity)
+	nodeInfo.addPod(podWithoutAffinity)
+	if !reflect.DeepEqual(nodeInfo.PodsWithAffinity(), []*v1.Pod{podWithAffinity}) {
+		t.Fatalf("get nodeinfo node error,expected: %v,got: %v", []*v1.Pod{podWithAffinity}, nodeInfo.PodsWithAffinity())
+	}
+}
+func TestAllowedPodNumber(t *testing.T) {
+	var nodeInfo *NodeInfo
+	if !reflect.DeepEqual(nodeInfo.AllowedPodNumber(), 0) {
+		t.Fatalf("get nodeinfo allowedpodnumber error,expected: %v,got: %v", 0, nodeInfo.AllowedPodNumber())
+	}
+
+	node := makeNode("testnode", map[string]string{})
+	nodeInfo = makeNodeInfo(node, []v1.Taint{})
```

Comment:
Please add meaningful `node.Status` and then checked the result, e.g. set `v1.ResourcePods` in `node.Status.Allocatable`.